object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -352,336 +332,34 @@ public Void visitExecutable(AnnotatedTypeMirror.AnnotatedExecutableType t, Void
       String methodName = element.getSimpleName().toString();
 
       TypeElement enclosingElement = (TypeElement) element.getEnclosingElement();
-      TypeMirror superclass = enclosingElement.getSuperclass();
 
       if ("toBuilder".equals(methodName)) {
-        if (hasAnnotation(enclosingElement, AutoValue.class)
-            && element.getModifiers().contains(Modifier.ABSTRACT)) {
-          handleToBuilder(t, enclosingElement, BuilderKind.AUTO_VALUE);
-          return super.visitExecutable(t, p);
-        }
-        // check superclass, to handle generated code
-        if (!superclass.getKind().equals(TypeKind.NONE)) {
-          TypeElement superElement = TypesUtils.getTypeElement(superclass);
-          if (hasAnnotation(superElement, AutoValue.class)) {
-            handleToBuilder(t, superElement, BuilderKind.AUTO_VALUE);
-            return super.visitExecutable(t, p);
-          }
-        }
-
-        if (hasAnnotation(element, "lombok.Generated")
-            || hasAnnotation(enclosingElement, "lombok.Generated")) {
-          handleToBuilder(t, enclosingElement, BuilderKind.LOMBOK);
+        for (FrameworkSupport frameworkSupport : frameworkSupports) {
+          frameworkSupport.handleToBulder(t);
         }
       }
 
+      // ensure nextEnclosingElement nonnull
       Element nextEnclosingElement = enclosingElement.getEnclosingElement();
       if (!nextEnclosingElement.getKind().isClass()) {
         return super.visitExecutable(t, p);
       }
 
-      BuilderKind builderKind = BuilderKind.NONE;
-
-      if (hasAnnotation(enclosingElement, AutoValue.Builder.class)) {
-        builderKind = BuilderKind.AUTO_VALUE;
-        assert hasAnnotation(nextEnclosingElement, AutoValue.class)
-            : "class " + nextEnclosingElement.getSimpleName() + " is missing @AutoValue annotation";
-
-      } else if ((hasAnnotation(enclosingElement, "lombok.Generated")
-              || hasAnnotation(element, "lombok.Generated"))
-          && enclosingElement.getSimpleName().toString().endsWith("Builder")) {
-        builderKind = BuilderKind.LOMBOK;
-      }
-
-      if (builderKind != BuilderKind.NONE) {
-
-        if (isBuilderBuildMethod(element, builderKind, nextEnclosingElement)) {
-          // determine the required properties and add a corresponding @CalledMethods annotation
-          Set<String> avBuilderSetterNames = getAutoValueBuilderSetterMethodNames(enclosingElement);
-          List<String> requiredProperties =
-              getRequiredProperties(nextEnclosingElement, avBuilderSetterNames, builderKind);
-          AnnotationMirror newCalledMethodsAnno =
-              createCalledMethodsForProperties(
-                  requiredProperties, avBuilderSetterNames, builderKind);
-          t.getReceiverType().addAnnotation(newCalledMethodsAnno);
-        }
+      for (FrameworkSupport frameworkSupport : frameworkSupports) {
```

Comment:
We can get rid of the negation in the previous `if` conditional, and then move this loop inside the `if`, replacing the `return` statement